kore/src/Kore/Internal/Alias.hs

```diff
@@ -25,40 +25,64 @@ import Kore.Debug
 import Kore.Internal.ApplicationSorts
 import Kore.Sort
 import Kore.Syntax.Application
+import Kore.Syntax.Variable
+       ( Variable )
 import Kore.Unparser
 
-data Alias =
+data Alias patternType =
     Alias
         { aliasConstructor :: !Id
         , aliasParams      :: ![Sort]
         , aliasSorts       :: !ApplicationSorts
+        , aliasLeft        :: [Variable]
+        , aliasRight       :: patternType
         }
     deriving (GHC.Generic, Show)
 
-instance Eq Alias where
+instance Eq patternType => Eq (Alias patternType) where
     (==) a b =
             Function.on (==) aliasConstructor a b
         &&  Function.on (==) aliasParams a b
+        &&  Function.on (==) aliasSorts a b
+        &&  Function.on (==) aliasLeft a b
+        &&  Function.on (==) aliasRight a b
     {-# INLINE (==) #-}
 
-instance Ord Alias where
+instance Ord patternType => Ord (Alias patternType) where
     compare a b =
             Function.on compare aliasConstructor a b
         <>  Function.on compare aliasParams a b
-
-instance Hashable Alias where
-    hashWithSalt salt Alias { aliasConstructor, aliasParams } =
-        salt `hashWithSalt` aliasConstructor `hashWithSalt` aliasParams
-
-instance NFData Alias
-
-instance SOP.Generic Alias
-
-instance SOP.HasDatatypeInfo Alias
-
-instance Debug Alias
-
-instance Unparse Alias where
+        <>  Function.on compare aliasSorts a b
+        <>  Function.on compare aliasLeft a b
+        <>  Function.on compare aliasRight a b
+
+instance Hashable patternType => Hashable (Alias patternType) where
+    hashWithSalt
+        salt
+        Alias
+            { aliasConstructor
+            , aliasParams
+            , aliasSorts
+            , aliasLeft
+            , aliasRight
+            }
+      =
+        salt
+        `hashWithSalt` aliasConstructor
+        `hashWithSalt` aliasParams
+        `hashWithSalt` aliasSorts
```

Comment:
We should not include `aliasSorts`, `aliasLeft`, or `aliasRight` because the combination of `aliasConstructor` and `aliasParams` uniquely determines these.